qiskit_aer/quantum_info/states/aer_statevector.py

```diff
@@ -0,0 +1,273 @@
+
+
+class AerStatevector(Statevector):
+    """AerStatevector class"""
+
+    def __init__(self, data, **configs):
+        """Initialize a statevector object.
+
+        Args:
+            data (np.array or list or Statevector or QuantumCircuit or qiskit.circuit.Instruction):
+                Data from which the statevector can be constructed. This can be either a complex
+                vector, another statevector or a ``QuantumCircuit`` or ``Instruction``
+                (``Operator`` is not supportted in the current implementation).  If the data is
+                a circuit or instruction, the statevector is constructed by assuming that all
+                qubits are initialized to the zero state.
+            configs (kwargs): configurations of ``AerState``. All the keys are handeld as string
+
+        Raises:
+            AerError: if input data is not valid.
+
+        Additional Information:
+            The ``dims`` kwarg is used to ``Statevector`` constructor.
+
+        """
+        if 'dims' in configs:
+            dims = configs.pop('dims')
+        else:
+            dims = None
+
+        if isinstance(data, (QuantumCircuit, Instruction)):
+            data, aer_state = AerStatevector._from_instruction(data, None, **configs)
+        elif isinstance(data, list):
+            data, aer_state = AerStatevector._from_ndarray(np.array(data, dtype=complex), **configs)
+        elif isinstance(data, np.ndarray):
+            data, aer_state = AerStatevector._from_ndarray(data, **configs)
+        elif isinstance(data, AerStatevector):
+            aer_state = data._aer_state
+            if dims is None:
+                dims = data._op_shape._dims_l
+            data = data._data
+        else:
+            raise AerError(f'Input data is not supported: type={data.__class__}, data={data}')
+
+        super().__init__(data, dims=dims)
+        self._aer_state = aer_state
+        self._result = None
+        self._configs = configs
+
+    def _last_result(self):
+        if self._result is None:
+            self._result = self._aer_state.last_result()
+        return self._result
+
+    @property
+    def metadata(self):
+        """Return result metadata of an operation that executed lastly."""
+        if self._last_result() is None:
+            raise AerError('AerState was not used and metdata does not exist.')
+        return self._last_result()['metadata']
+
+    def __deepcopy__(self, _memo=None):
+        data, aer_state = AerStatevector._from_instruction(
+            QuantumCircuit(self._aer_state.num_qubits),
+            self._data, **self._configs)
+        return AerStatevector(data, _aer_state=aer_state, **self._configs)
+
+    def __eq__(self, other):
+        eq_quantumqtate = isinstance(other, Statevector) and self.dims() == other.dims()
+        eq_statevector = np.allclose(self._data, other._data, rtol=self.rtol, atol=self.atol)
+        return eq_quantumqtate and eq_statevector
```

Comment:
Generating both booleans and _then_ doing `x and y` bypasses the shortcircuiting, which means if `other` isn't a `Statevector`, then the second line will raise an `AttributeError` on `other._data`.  I'm not sure you need to override this method at all, though?  It just does the same thing as the base class's.